### ISSUE TITLE
Plotting cue accuracies for test data

### ISSUE THREAD

**Issue by hneth** [ndphillips/FFTrees#63]
When creating an `FFTrees` object with both training and test data, I am unable to plot the cue accuracies for the **test** data.  

For example, the following code only yields 

`Error in showcues(x = x, data = data, main = main) :`   
`    There are no test statistics in this object.`

```{r example}
# Create a fast-and-frugal tree (FFT) predicting heart disease:
heart.fft <- FFTrees(formula = diagnosis ~.,
                     data = heart.train,
                     data.test = heart.test,
                     main = "Heart Disease",
                     decision.labels = c("Healthy", "Diseased"))

# Visualize cue accuracies for test data:
plot(heart.fft, what = "cues", data = "test")
```

However, the `data` argument of `showcues()` suggests that we can plot cue accuracies for test data... 

Perhaps this is intentional (as the cue accuracies during training may be more important than those in testing?), but I also could not find these cue accuracies for test data anywhere.  If they are not a default part of the `FFTrees` object, perhaps they can be created upon request?  

After all, the correspondence between cue accuracies in training and testing should be an interesting predictor for performance decrements from fitting to predicting...

**Comment by ndphillips** [comment]
Indeed this is bad documentation. There currently is no method to calculate cue accuracies on test data. We either need to add this or remove the `data` argument from `showcues()`

**Comment by hneth** [comment]
I realize that computing cue accuracies for test data can be tricky.  

First, its not immediately obvious, whether this should

- (a) calculate new cue directions and decision thresholds (for the test data, given the current goal), or 
- (b) apply the cue directions and decision thresholds computed for the training data on the test data.

Second, it may be overkill to always compute and store information that is rarely needed and unnecessarily bloat the corresponding`FFTrees` object with it. 

Thus, unless there are better reasons or future requests for this feature, I'd suggest removing the `data` argument from `showcues()` and update its documentation accordingly.  If you agree, calling `showcues()` via `FFTrees(x, what = "cues")` should probably include a check and provide a feedback message when `data = "test"` is simultaneously specified?

**Comment by ndphillips** [comment]
Totally agree with your points here. I agree that as a first step the `data` argument can be removed from `showcues()`. Can think later if we want to actually implement that functionality

**Comment by hneth** [comment]
When starting to look into this, I spotted another issue with the `showcues()` function:  

The current version initially reads `goal <- x$params$goal`, but instead of using this later, it always sorts cue accuracies by "wacc" (as `cue.df$wacc` is hard-coded when ranking `cue.df`).  

I suggest generalizing that to ranking by `goal`, so that it becomes possible to plot (training) cue accuracies for alternative goals.  (This will only matter when the goal is being changed away from "wacc".  But as this seems legitimate, the function should accommodate the current goal, rather than enforcing a single one.)

**Comment by hneth** [comment]
Latest commits fix both bugs (regarding the use of `data` and current `goal` settings).  Additional `alt.goal` and `quiet` arguments increase functionality and provide corresponding user feedback.

**Comment by hneth** [comment]
As the corresponding commits have been merged (in #71), I'll close this issue now.  

Nevertheless, we may want to revisit the issue of computing and plotting cue accuracies for test data at some future point in time.
